docs/src/main/mdoc/migration.md

```diff
@@ -0,0 +1,52 @@
+---
+layout: home
+title:  "migration guide"
+section: "Home"
+position: 7
+---
+
+## v0.18.0 Migraion Guide
+
+Spire supports Scala 3 since version 0.18.0!
+
+If you find any erroneous behavior due to Scala 3 compatibility, please feel free to open issue or send PR!
+
+Spire tries keeping compatibility as much as possible, but some APIs change.
+If your code uses following APIs, you need to modify them.
+
+## API changes
+ 
+### General
+- because the method `Dist.given` clashes with the `given` keyword, it was renamed to `given_`
+- the class `DistIterator` has a `next` member that clashes with a `next` method. The field has renamed to `_next`
+- `UnboundSyntax` seems to be bogus and it is not used anywhere in `spire`. It was marked as deprecated in scala 2 and removed in scala 3
+- parentheses: these are now omitted for 0-arity (empty-parens) methods with no side-effects.
+### Macro
+- `Checked.tryOrReturn` was removed.
+- Use of `Machinist` was removed.
+- `PackMacro` is available both Scala 2 and 3.
+- `cFor` was deprecated and renamed to `fastFor`. Available both Scala 2 (as `cFor`) and 3(as `fastFor`). Performance-sensitive code should continue using `cfor` on Scala 2.
```

Comment:
There are some important explanations for this, it would be strange if the release notes are more detailed than the migration guide 😆